File: backend/PxlTeambuilderApi/PxlTeambuilderApi.Data/Domain/Project.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Text;
using PxlTeambuilderApi.Data.Interfaces;

namespace PxlTeambuilderApi.Data.Domain
{
    public class Project: IJoinable
    {
        [JsonProperty("projectId")]
        public string ProjectId { get; set; }

        [JsonProperty("userId")]
        public int UserId { get; set; }

        [JsonProperty("title")]
        public string Title { get; set; }

        [JsonProperty("description")]
        public string Description { get; set; }

        [JsonProperty("maxstudentspergroup")]
        public int MaxStudentsPerGroup { get; set; }

        //navigation properties
        [JsonIgnore]
        public virtual ICollection<Group> Groups { get; set; }

        [JsonIgnore]
        public virtual ICollection<UserProjectDetail> UserProjectDetails { get; set; }

        [JsonIgnore]
        public virtual User Creator { get; set; }
    }
}

File: frontend/pxlTeambuilder/src/app/models/project.ts
export class Project{

    private id : string;
    private userId : number;
    private title : string;
    private description : string;
    private maxStudentsPerGroup : number;

    constructor(){}

    public set Id(id : string){
        this.id = id;
    }

    public set UserId(userId : number){
        this.userId = userId;
    }

    public set Title(title : string){
        this.title = title;
    }

    public set Description(description : string){
        this.description = description;
    }

    public set MaxStudentsPerGroup(maxStudentsPerGroup : number){
        this.maxStudentsPerGroup = maxStudentsPerGroup;
    }

    public get Id(){
        return this.id;
    }

    public get UserId(){
        return this.userId;
    }

    public get Title(){
        return this.title;
    }

    public get Description(){
        return this.description;
    }

    public get MaxStudentsPerGroup(){
        return this.maxStudentsPerGroup;
    }
}